### PR TITLE
Shears setting

### PR DESCRIPTION
**Pull request by ImEr1c**
Closes #
Closes #
Closes #

# Changes proposed in this pull request:
-Adds a new setting to the Forester, Builder and miner to allow players to choose if they want to use shears or not
-
-

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/util/WorkerUtil.java [hunk 1]
```diff
@@ -165,11 +165,11 @@ public static boolean setSpawnPoint(@Nullable final BlockPos spawnPoint, @NotNul
      * @param blockHardness the hardness.
      * @return the toolType to use.
      */
-    public static IToolType getBestToolForBlock(final BlockState state, float blockHardness)
+    public static IToolType getBestToolForBlock(final BlockState state, float blockHardness, final boolean canUseShears)
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/util/WorkerUtil.java]: I thought about the implementation of this, but this is very inflexible.
a) Hand the worker building through
b) Have only one global setting for this (store the key in the AbstractBuilding)
c) Here we then query getOptionalSetting with the shears key from the building

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 2]
```diff
@@ -1244,20 +1246,7 @@ private void requestTool(@NotNull final BlockState target, final BlockPos pos)
...
-        {
-            return building.getSetting(BuildingLumberjack.USE_SHEARS).getValue();
-        }
-        else if (building instanceof BuildingMiner)
-        {
-            return building.getSetting(BuildingMiner.USE_SHEARS).getValue();
-        }
-
-        return false;
+        return building.getOptionalSetting(USE_SHEARS).orElse(new BoolSetting(false)).getValue();
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java]: move this call into the method that checks for the tools and remove canUseShears method entirely then

Comment (ImEr1c) [reply src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java]: There are 2 methods that need this: getMostEfficientTool and requestTool

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java]: You can either hand the AI that will handle it, or the building, and then query it at the place you need it. 
The canUseShears should not be a method param.

Comment (ImEr1c) [reply src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java]: Is it ok now?

File: src/main/java/com/minecolonies/coremod/util/WorkerUtil.java [hunk 3]
```diff
@@ -165,11 +168,11 @@ public static boolean setSpawnPoint(@Nullable final BlockPos spawnPoint, @NotNul
      * @param blockHardness the hardness.
      * @return the toolType to use.
      */
-    public static IToolType getBestToolForBlock(final BlockState state, float blockHardness)
+    public static IToolType getBestToolForBlock(final BlockState state, float blockHardness, final AbstractBuilding building)
     {
         final net.minecraftforge.common.ToolType forgeTool = state.getHarvestTool();
 
-        if (state.getBlock() instanceof IForgeShearable)
+        if (building.getOptionalSetting(USE_SHEARS).orElse(new BoolSetting(false)).getValue() && state.getBlock() instanceof IForgeShearable)
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/util/WorkerUtil.java]: Looks fine, only thing I'd change would be default to true.

Comment (ImEr1c) [reply src/main/java/com/minecolonies/coremod/util/WorkerUtil.java]: done

File: src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingMiner.java [hunk 4]
```diff
@@ -14,6 +14,7 @@
 import com.minecolonies.coremod.colony.buildings.AbstractBuildingStructureBuilder;
 import com.minecolonies.coremod.colony.buildings.modules.WorkerBuildingModule;
 import com.minecolonies.coremod.colony.buildings.modules.settings.BlockSetting;
+import com.minecolonies.coremod.colony.buildings.modules.settings.BoolSetting;
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingMiner.java]: needs some cleanup on he imports now (also builder)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java [hunk 5]
```diff
@@ -29,6 +29,10 @@
 import com.minecolonies.api.util.constant.translation.RequestSystemTranslationConstants;
 import com.minecolonies.coremod.colony.buildings.AbstractBuilding;
 import com.minecolonies.coremod.colony.buildings.modules.WorkerBuildingModule;
+import com.minecolonies.coremod.colony.buildings.modules.settings.BoolSetting;
+import com.minecolonies.coremod.colony.buildings.workerbuildings.BuildingBuilder;
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java]: also cleanup here

Comment (ImEr1c) [reply src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java]: done

**Review by Raycoms — DISMISSED** [review]

**Comment by batmanatorul** [comment]
Im confused, why is it still requesting shears? I thought this would solve it..

**Comment by batmanatorul** [comment]
Im confused, why is the lumberjack still requesting shears? I thought that would solve it

**Comment by ImEr1c** [comment]
Review?

**Review by Raycoms — APPROVED** [review]

**Review by someaddons — APPROVED** [review]

**Pull request merged**